### ISSUE TITLE
time mismatch between dynamics and MB Mod

### ISSUE THREAD

**Issue by fmaussion** [OGGM/oggm#1020]
Fun fact: there is an issue in how we report MB volume changes in OGGM at the moment. It's not a big deal and probably easy to fix, but still falls into the "WTF" category haha.

Explanation: the way we see OGGM output files is that they are snapshots, i.e. the written volume on timestamp year 2003 is the status of the glacier on Oct 2003 (hydrological year convention), and the timestamp 2004 is Oct 2004. 

However, the dynamical model does not really care about real time. If we start in year 2003, it will ask for mass-balance at fractional year 2003.0, 2003.1, .... etc until 2004.0. 

So what we currently store with the timestamp 2004 is the glacier evolution under the mass-balance year 2003!!!!

(noticed this while playing around with a mass-balance analysis tutorial :facepalm: )

**Comment by fmaussion** [comment]
The consequences are:
- we need to fix this (quickly)
- the consequences for all of you are quite limited, because we are talking about a one-year shift basically. The problem is only striking when comparing OGGM dynamical runs MBs with WGMS 

What's interesting to learn from this is that except for @juliaeis , I think that nobody ever really checked the differences between the fixed geometries MBs and the varying geometries MBs. 

I noticed while writing a tutorial about exactly this

**Comment by fmaussion** [comment]
**FALSE ALARM** 
(or, I should rather say: stay alert)

I got bitten by my own decisions. There is no "bug" in the timestamp, there is rather a problem of interpretation of the timestamp. 

What we call the "hydrological year" in OGGM is following the usual convention, i.e. Oct 2002 to Sept 2003. Within OGGM we have the tools to play with that:

```python
from oggm.utils import floatyear_to_date, hydrodate_to_calendardate
print('Hydro dates')
print(floatyear_to_date(2003.0))
print(floatyear_to_date(2003.99))
print('Calendar dates')
print(hydrodate_to_calendardate(2003, 12, start_month=10))
print(hydrodate_to_calendardate(2003, 1, start_month=10))
```

outputs

```
Hydro dates
(2003, 1)
(2003, 12)
Calendar dates
(2003, 9)
(2002, 10)
```

Which is correct. Also the calendar time stamp in the OGGM files is correct: I just forgot to look at it and panicked instead.

So, long story short:
- WE HAVE TO BE CAREFUL WHEN REPORTING OUR DATA TO OTHERS (calendar dates do not lie)
- I still have to change one thing in OGGM: the RGI dates have to be converted to hydro dates for the run, i.e. the Alps which have an RGI timestamp "2003" should have their simulations started in October 2003, i.e. hydro timestamp 2004!!!
